ex35.py

```diff
@@ -6,9 +6,11 @@ def gold_room():
 
     choice = raw_input("> ")
     try:
-        if re.match('[0-9]', choice) and int(choice) < 50:
+        goldnum = int(choice)
+        if goldnum < 50:
             dead("Nice, you're not greedy, you win!")
-        elif int(choice) >= 50:
+        # else if goldnum is *not* less than 50..
+        elif not (goldnum < 50):

```

Comment:
Looking at this now, I see I should have just replaced the second test here with an "else".